components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2742,7 +2743,10 @@ private FederatedAuthenticatorConfig updateFederatedAuthenticatorConfig(String f
                   FederatedAuthenticatorPUTRequest authenticator) throws IdentityProviderManagementClientException {
 
         String authenticatorName = getDecodedAuthenticatorName(federatedAuthenticatorId);
-        DefinedByType definedByType = resolveDefinedByTypeToUpdateFederatedAuthenticator(authenticatorName);
+        /* As this operation is only for updating an existing authenticator, the definedByType parameter in the PUT
+         request is ignored. */
+        DefinedByType definedByType = resolveDefinedByTypeToUpdateFederatedAuthenticator(authenticatorName,
+                null);
```

Comment:
why pass null and get the definedByType.

If not considering rely on the existing ones value itself

Reply:
Also, isn't this method is used within updating federated authenticators in IdP.
In such case, suppose a user defined federated authenticator is added over a PUT request. Won't that go over this path